### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by ashhhleyyy**
~~not tested yet~~ now tested
includes a breaking change that needs to be made in nucleoid-extras at the same time:
- remove discriminator from DiscordUser (used in discord -> minecraft chat relay) (fixed by NucleoidMC/nucleoid-extras#200)

### CODE REVIEW

File: src/discord.rs [hunk 1]
```diff
@@ -81,147 +100,128 @@ pub async fn run(controller: Address<Controller>, config: DiscordConfig) {
...
-                        w.embeds(vec![embed]);
-                        w.username("Backend error reporting");
-                        w
-                    })
-                    .await
-                {
+                let embed = CreateEmbed::new()
+                    .title(message.title)
+                    .description(message.description)
+                    .fields(if let Some(fields) = message.fields {
```
Comment (Gegy) [inline src/discord.rs]: Nit: maybe something like `message.fields.into_iter().flatten()`?

File: src/discord/lfp.rs [hunk 2]
```diff
@@ -118,7 +118,7 @@ struct Registration {
 }
 
 pub struct Handler {
-    pub discord: Address<DiscordClient>,
+    pub _discord: Address<DiscordClient>,
```
Comment (Gegy) [inline src/discord/lfp.rs]: Nit: I forget if there was a reason to have these here, but maybe if they are unused we can fully drop them?

**Review by Gegy — APPROVED** [review]
Awesome! Looks like some really nice cleanups, thanks for figuring out all those upgrades! \o/
Just had some tiny nits, but totally optional :)

File: src/discord/lfp.rs (hunk 2 above)
Comment (ashhhleyyy) [reply src/discord/lfp.rs]: i couldn't see any reason for keeping them around other than symmetry with the other handler struct in pings. i tested removing them and nothing seems to break

**Pull request merged**